### PR TITLE
Local env store: deny access to certain env vars

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Changes the local env secret store to disallow access to certain variables used by Dapr's security features:

- APP_API_TOKEN
-  DAPR_API_TOKEN
- DAPR_TRUST_ANCHORS
- DAPR_CERT_CHAIN
- DAPR_CERT_KEY

### CODE REVIEW

**Review by berndverst — APPROVED** [review]

File: secretstores/local/env/envstore.go [hunk 1]
```diff
@@ -77,3 +84,13 @@ func (s *envSecretStore) GetComponentMetadata() map[string]string {
 	metadata.GetMetadataInfoFromStructType(reflect.TypeOf(metadataStruct), &metadataInfo)
 	return metadataInfo
 }
+
+func (s *envSecretStore) isKeyAllowed(key string) bool {
+	switch key {
+	case "APP_API_TOKEN", "DAPR_API_TOKEN",
```
Comment (berndverst) [inline secretstores/local/env/envstore.go:90]: I would really prefer if `dapr/kit` provided these reserved environment variables.

But for now this is fine.

**Comment by berndverst** [comment]
@ItalyPaleAle please document this.

**Comment by ItalyPaleAle** [comment]
@berndverst  do you think this needs to be documented? Users shouldn't really be concerned with these env vars that are internal IMHO

**Comment by yaron2** [comment]
Not sure this needs to be documented.

**Comment by berndverst** [comment]
How about an error message that says accessing this Dapr internal environment variable is disallowed?

**Review by berndverst — CHANGES_REQUESTED** [review]
Add an error message that states that accessing Dapr internal environment variables is disallowed.

**Comment by berndverst** [comment]
`/ok-to-test`

**Comment by ItalyPaleAle** [comment]
Added warning log

**Review by berndverst — APPROVED** [review]

**Pull request merged**